### ISSUE TITLE
[unbound-method] False positive on `Intl.Collator.compare`

### ISSUE THREAD

**Issue by kestrelnova** [typescript-eslint/typescript-eslint#4383]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```json
{
  "rules": {
    "@typescript-eslint/unbound-method": "error"
  }
}
```

```ts
const collator = new Intl.Collator("en")
const sorted = ["foo", "bar"].sort(collator.compare)
```

(tsconfig not included as the same issue occurs regardless of config)

**Expected Result**

No error, as `Intl.Collator.compare` is a native bound method intended to be used this way.

**Actual Result**

```console
2:36  error  Avoid referencing unbound methods which may cause unintentional scoping of `this`.
If your function does not access `this`, you can annotate it with `this: void`, or consider using an arrow function instead  @typescript-eslint/unbound-method
```

**Additional Info**

[The rule's source code](https://github.com/typescript-eslint/typescript-eslint/blob/ccee1a204bcca0e6a32b219076ed4fce8880bff4/packages/eslint-plugin/src/rules/unbound-method.ts#L21-L100) does account for builtins, but it looks like only top-level bound methods are put on the list to ignore when checking. So `Intl.Collator` itself is ignored, but `Intl.Collator.compare` isn't because it's another level down. Presumably the same issue affects other similar methods like `Intl.DateFormat.format` as well -- maybe check the `Intl` constructors individually when generating the ignore list?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.8.1` |
| `@typescript-eslint/parser`        | `5.8.1` |
| `TypeScript`                       | `4.5.4` |
| `ESLint`                           | `8.6.0` |
| `node`                             | `17.3.0` |

**Comment by bradzacher** [comment]
[I am unable to repro this](https://typescript-eslint.io/play/#ts=4.5.2&sourceType=module&code=PQKgBApgzgNglgOwC5gEQAEkE8AO0DGATnDkgLTTzLACuCARgPZ0AmZAthEgBaMuoAuNBEKFGhVGBDAAsACh5+RgigolMGAEMk4sAF4wCCAHcwASWQwAdAGFGG7eIAUqCAlQBKRctVgo4pAgWfTAAbVQAM0ZGVAAaNHpNCQBdK39CJCd1LR1CKyV2HCSILzkgA&rules=N4XyA&tsConfig=N4XyA)

Please use the playground to create an isolated repro so we can investigate further
